src/_tutorials/language/futures.md

```diff
@@ -178,30 +179,41 @@ daily news digest is printed.
 The following diagram shows the flow of execution through the code.  Each
 number corresponds to a step below.
 
-<img src="../images/async-await.png"
-     alt="diagram showing flow of control through the main() and printDailyNewsDigest functions" />
+<!--img src="../images/async-await.png"
+     alt="diagram showing flow of control through the main() and printDailyNewsDigest functions" /-->
+<!-- this diagram needs to be updated -->
 
 1. The app begins executing.
-1. The main function calls `printDailyNewsDigest()`, which (because it's
-   marked `async`), immediately returns a Future, _before any code is
-   executed_.
+1. The main function calls `printDailyNewsDigest()`, which begins executing
+   synchronously.
+1. `printDailyNewsDigest()` calls `gatherNewsReports()`, which also begins
+   executing synchronously.
+1. `gatherNewsReports()` will be guaranteed to return a Future -- either uponthe
+   first `await`, thrown exception, return, or at the end. In this case, rather
+   than returning `newsStream.first`, a `Future<String>` is returned and
+   `gatherNewsReports()` pauses.
+1. The `Future` from `gatherNewsReports()` hits the await expression
+   (`await gatherNewsReports()`) and the program pauses, waiting for that Future
+   to complete.
+1. Like `gatherNewsReports()`, `printDailyNewsDigest()` is async. Since this is
+   its first `await`, it pauses and returns a `Future<void>` to its caller
+   (`main`).
 1. The remaining print functions execute. Because they're synchronous, each
    function executes fully before moving on to the next print function. For
    example, the winning lottery numbers are all printed before the weather
    forecast is printed.
-1. The body of the `printDailyNewsDigest()` function starts executing.
-
-1. After reaching the await expression (`await gatherNewsReports()`) and
-   calling `gatherNewsReports()`, the program pauses, waiting for the Future
-   returned by `gatherNewsReports()` to complete.
-1. Once that Future completes, execution of `printDailyNewsDigest()`
-   continues, printing the news.
+1. Once the Future returned by `gatherNewsReports()` completes (in this case, it
+   will complete immediately), execution of `printDailyNewsDigest()` continues,
+   printing the news.
 1. When the `printDailyNewsDigest()` function body has completed executing,
    the Future that it originally returned completes, and the app exits.
 
 <aside class="alert alert-info" markdown="1">
-  **Note:** If an async function doesn't explicitly return a value, it returns
-  a Future wrapped around a null value.
+Note the return types carefully. Instead of `gatherNewsReports()` returning a
```

Comment:
Good stuff, but I have a bit of rewriting (because I'm me, and to avoid using method names as verbs). How about:

Note the return types. The return type of `gatherNewsReports()` is `Future<String>`, which means that it returns a Future that completes with a String value. The `printDailyNewsDigest()` function, which doesn't explicitly return a value, has the return type `Future<void>`.